source/library/Wuss.hs

```diff
@@ -152,20 +158,21 @@ defaultConfig = do
 
 -- | Runs a secure WebSockets client with the given 'Config'.
 runSecureClientWithConfig
-  :: Socket.HostName -- ^ Host
+  :: UnliftIO.MonadUnliftIO m
+  => Socket.HostName -- ^ Host
   -> Socket.PortNumber -- ^ Port
   -> String.String -- ^ Path
   -> Config -- ^ Config
   -> WebSockets.ConnectionOptions -- ^ Options
   -> WebSockets.Headers -- ^ Headers
   -> WebSockets.ClientApp a -- ^ Application
-  -> IO.IO a
+  -> m a
 runSecureClientWithConfig host port path config options headers app = do
-  context <- Connection.initConnectionContext
-  Exception.bracket
-    (Connection.connectTo context (connectionParams host port))
-    Connection.connectionClose
-    (\connection -> do
+  context <- MonadIO.liftIO Connection.initConnectionContext
+  UnliftIO.bracket
```

Comment:
Could this use `Control.Monad.Catch.bracket` and `Control.Monad.IO.Class.liftIO` instead? Then I wouldn't need to depend on `unliftio`, and the constraint would be `MonadIO m, MonadThrow m`.

Reply:
MonadMask is needed for bracket, but sure

Reply:
The downside being that it won't work with unliftio in the same way ofc